### ISSUE TITLE
WCAG 3.3.2 A: Required fields not indicated (Issue 32)

### ISSUE THREAD

**Issue by CJE001** [civictheme/uikit#361]
### Summary

Via Vision Australia assessment: August 2024

Impact: medium

Note: DTA have a 90 day remediation period to address the identified issues within the audit, all issues must be resolved to obtain WCAG 2.2 certification for digital.gov.au

### Steps to reproduce

[Contact digital.gov.au beta | digital.gov.au](https://www.digital.gov.au/contact) (https://www.digital.gov.au/contact)

### Observed outcome

The "Subject" and "Message" inputs of the contact form are required fields. While this information is available programmatically for assistive technologies, this information is not provided in text. 

![Picture12](https://github.com/user-attachments/assets/b4285b6c-dcd6-4f68-990b-620be0ccd838)

### Code used

```
<input type="text" value="test" class="ct-textfield ct-input__element ct-theme-light form-text required" #component_theme="light" data-drupal-selector="edit-subject" id="edit-subject" name="subject" size="60" maxlength="255" required="required" aria-required="true">
…
<textarea class="ct-textarea ct-input__element ct-theme-light ct-input--required required " rows="5" required="" #civictheme_theme="light" #component_theme="light" data-drupal-selector="edit-message" id="edit-message" name="message" cols="60" aria-required="true"></textarea>
```
### Why this matters

When required fields are not specified, users may struggle to complete forms successfully. This can be especially problematic for users with cognitive or learning disabilities. 

### Expected outcome

Ensure that users are advised in text when a form contains required fields. 

This can be achieved in a number of ways such as:

•	Appending the text (required) to all `<label>` elements of required inputs
•	Appending a symbol, such as an asterisk `(*)` to all `<label>` elements of required inputs, and explaining the meaning of this asterisk above the form
•	Appending the text (optional) to all `<label>` elements of optional inputs, where all others are then required. 

### Code example

```
<div>
<label for="subject">
Subject (required)
</label>
</div>
<div>
<input type="text" value="…" id="subject" name="subject" required aria-invalid="false" aria-describedby="error-1">
</div>
```

### Related issues

For other issues related to this component, please refer to:

•	Issue 6 – Label not programmatically associated with input
•	Issue 7 – Error messages not programmatically associated with input
•	Issue 12 – Inputs missing autocomplete attribute
•	Issue 20 – Form input borders have insufficient contrast 
•	Issue 37 – Success message not announced by screen readers

**Comment by CJE001** [comment]
**Guideline 3.3: Input Assistance "Help users avoid and correct mistakes."

3.3.2 Labels or Instructions WCAG 2.0 Level A**

[Labels](http://www.w3.org/TR/WCAG20/#labeldef) or instructions are provided when content requires user input.
Users with a disability are more likely to make mistakes than others, and recovery from mistakes may be more difficult. Providing simple instructions and examples of expected data format help all users, and especially people with a disability, fill out forms correctly.

**Comment by daniel-grynenko** [comment]
**Test Env:** DEV
**Test Status:** PASS
**Test Result:**
1. Verified that required fields in the contact form now include a visible indicator next to the label
2. Confirmed that all fields configured as required will visually display their required status in text format
3. Ensured that screen readers correctly recognise and announce the required status of the fields
4. Validated that the form submission process correctly enforces required fields

Required attribute is present in the code and as visible text 
<img width="600" alt="Image" src="https://github.com/user-attachments/assets/f5276fd1-c1b1-441a-be4f-377fa2d5ef46" />

Screen reader output
<img width="600" src="https://github.com/user-attachments/assets/040895c4-d903-44cb-8178-3adf463d1f4b" />
